dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitorTest.java

```diff
@@ -0,0 +1,191 @@
+import org.elasticsearch.common.settings.Settings;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static com.dotcms.util.CollectionsUtils.list;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ESReadOnlyMonitorTest {
+
+    private static ESReadOnlyMonitor esReadOnlyMonitor;
+    private SystemMessageEventUtil systemMessageEventUtilMock;
+    private RoleAPI roleAPIMock;
+
+    @BeforeClass
+    public static void prepare () throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    @Before
+    public void init () {
+        systemMessageEventUtilMock = mock(SystemMessageEventUtil.class);
+        roleAPIMock = mock(RoleAPI.class);
+
+        esReadOnlyMonitor = ESReadOnlyMonitor.getInstance(systemMessageEventUtilMock, roleAPIMock);
+    }
+
+    /**
+     * Method to Test: {@link ESReadOnlyMonitor#start(ReindexEntry, String)}
+     * When: If the LIVE and WORKING current index are not read only
+     * Should: not sent large message
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldNotSendLargeMessage() throws DotDataException {
+        final ReindexEntry reindexEntry = mock(ReindexEntry.class);
+        final String cause = "cause";
+
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        putReadOnly(indiciesInfo.getWorking(), false);
+        putReadOnly(indiciesInfo.getLive(), false);
+
+        esReadOnlyMonitor.start(reindexEntry, cause);
+
+        verify(systemMessageEventUtilMock, never()).pushLargeMessage(any(), any());
+
+        assertEquals(false, ESIndexUtil.isReadOnly(indiciesInfo.getWorking(), indiciesInfo.getLive()));
+    }
+
+    /**
+     * Method to Test: {@link ESReadOnlyMonitor#start(ReindexEntry, String)}
+     * When: If any of LIVE and WORKING current index are read only
+     * Should: sent large message
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSendLargeMessage() throws DotDataException, DotSecurityException, InterruptedException {
+        final ReindexEntry reindexEntry = mock(ReindexEntry.class);
+        final String cause = "cause";
+
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final Role adminRole = mock(Role.class);
+        when(roleAPIMock.loadCMSAdminRole()).thenReturn(adminRole);
+
+        final User user = mock(User.class);
+        when(user.getUserId()).thenReturn("1");
+
+        when(roleAPIMock.findUsersForRole(adminRole)).thenReturn(list(user));
+
+        try {
+            putReadOnly(indiciesInfo.getWorking(), true);
+            putReadOnly(indiciesInfo.getLive(), false);
+
+            esReadOnlyMonitor.start(reindexEntry, cause);
+
+            Thread.sleep(100);
```

Comment:
is there another way to know when the monitor is ready? this sleep might cause a race condition